builtInFunctions/esdtMetaDataRecreate.go

```diff
@@ -152,18 +152,35 @@ func getEsdtInfo(
 		}
 
 		return &esdtStorageInfo{
-			esdtData:     esdtData,
-			esdtTokenKey: esdtTokenKey,
-			nonce:        nonce,
-			isDynamic:    true,
+			esdtData:            esdtData,
+			esdtTokenKey:        esdtTokenKey,
+			nonce:               nonce,
+			metaDataInSystemAcc: true,
 		}, nil
 	}
 
-	esdtData, err := storageHandler.GetESDTNFTTokenOnSender(acntSnd, esdtTokenKey, nonce)
+	esdtData, isNew, err := storageHandler.GetESDTNFTTokenOnDestination(acntSnd, esdtTokenKey, nonce)
 	if err != nil {
 		return nil, err
 	}
 
+	if tokenType == uint32(core.DynamicNFT) {
+		if isNew {
+			esdtData.TokenMetaData = &esdt.MetaData{}
+			esdtData.Type = tokenType
+		}
+		return &esdtStorageInfo{
+			esdtData:            esdtData,
+			esdtTokenKey:        esdtTokenKey,
+			nonce:               nonce,
+			metaDataInSystemAcc: false,
+		}, nil
+	}
+
+	if isNew {
```

Comment:
in which case do we get here ?

Reply:
In case an address has the `Update` role for a normal NFT, but it does not have the NFT.